http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -361,7 +364,11 @@ class GKEInterpreter[F[_]](
       ).compile.lastOrError
 
       //TODO: validate app release is the same as restore release
-      galaxyRestore <- persistentDiskQuery.getGalaxyDiskRestore(diskId).transaction
+      appRestore: Option[AppRestore] <- persistentDiskQuery.getAppDiskRestore(diskId).transaction
+      galaxyRestore: Option[GalaxyRestore] = appRestore.flatMap {
```

Comment:
Can you explain this code? It seems odd to have CromwellRestore be a case within something a block of code that seems specific to Galaxy.

Reply:
Since `getAppDiskRestore` now returns an `AppRestore`, it is trying to convert it into `GalaxyRestore`. So if it encounters a `CromwellRestore` for the disk, it returns `None`. But now I am thinking if it actually returned `CromwellRestore`, instead of returning `None` it show throw an error? 🤔

Reply:
I don't understand the syntax/architecture-- perhaps you can explain it to me during the mob time.